src/ElasticApm.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Drupal\elastic_apm;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+
+use PhilKra\Agent;
+
+/**
+ * The Elastic APM service object.
+ *
+ * @package Drupal\elastic_apm
+ */
+class ElasticApm implements ElasticApmInterface {
+
+  /**
+   * A config object for the elastic_apm configuration.
+   *
+   * @var \Drupal\Core\Config\Config
+   */
+  protected $config;
+
+  /**
+   * The current account.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  protected $account;
+
+  /**
+   * Constructs an Elastic APM object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   The config factory service.
+   * @param \Drupal\Core\Session\AccountProxyInterface $account
+   *   The current account.
+   */
+  public function __construct(
+    ConfigFactoryInterface $configFactory,
+    AccountProxyInterface $account
+  ) {
+    $this->config = $configFactory->get('elastic_apm.configuration');
+    $this->account = $account;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getAgent() {
+    // Initialize and return our PHP Agent.
+    return new Agent(
+      $this->getConfig(),
+      [
+        'user' => [
+          'id' => $this->account->id(),
+          'email' => $this->account->getEmail(),
+        ],
+      ]
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConfig() {
+    // Fetch the configs.
+    $elastic_apm_config = $this->config->get();
+    // Set the apmVersion to v1 if it's empty as the PHP Agent doesn't.
+    if (empty($elastic_apm_config['apmVersion'])) {
+      $elastic_apm_config['apmVersion'] = 'v1';
+    }
```

Comment:
**CHANGE REQUEST**

Can we make this a constant in this class i.e. `const DEFAULT_APM_VERSION = 'v1';`

However, since we do provide a default value upon installation it will only be missing by error and we want to know that. Maybe we should make this field mandatory in the form and include this in the `isConfigured` function as required config.

**CHANGE REQUEST**

*API*

This is not a public API function i.e. it doesn't seem to be needed outside of the class. It should therefore be marked as `protected`.

Reply:
As you've suggested, I actually don't see the need for the getConfig now. If we make the apmVersion required, we don't need to set the default value for it in the getConfig() function. So now, getConfig would just return the entire config as is, which would allow us to do this in the constructor:

$this->config = $configFactory->get('elastic_apm.configuration')->get();